app/models/health_care_application.rb

```diff
@@ -267,4 +267,28 @@ def log_submission_failure
   def send_failure_mail
     HCASubmissionFailureMailer.build(parsed_form['email'], google_analytics_client_id).deliver_now
   end
+
+  # If the hca_use_facilities_API flag is on then vaMedicalFacility will only
+  # validate for a string, else it will validate through the enum.  This avoids
+  # changes to vets-website and vets-json-schema having to deploy simultaneously.
+  def form_matches_schema
+    if form.present?
+      JSON::Validator.fully_validate(current_schema, parsed_form).each do |v|
+        errors.add(:form, v.to_s)
+      end
+    end
+  end
+
+  def current_schema
+    @current_schema ||= begin
+      schema = VetsJsonSchema::SCHEMAS[self.class::FORM_ID]
+      if Flipper.enabled?(:hca_use_facilities_API)
```

Comment:
using unless with a return would have been cleaner, however rubocop was unhappy with this and adding an override here was frankly more problematic in the future... potentially.

Reply:
🤔 Do you remember which rubocop complaint it was? `return [...] unless [...]` is such a common ruby pattern, that's a surprise to me.

I guess the only reasonable alternative here would be putting 284-290 into _another_ function that early-returns on its own, but that does seem excessive.